### PR TITLE
Improved initialization in Batch API

### LINKED ISSUE #282: Improve Batch API startup
So far, batch API has implemented a lazy startup. This method is really good when Batch API have a considerable amount of traffic, but nowadays Batch API is idle and after startup we have to implement a discover mechanism to consume job from queues.

### PR DESCRIPTION
**Pull request by dgaubert**
Now Batch API looks for queues with jobs to consume at service initiaization.

Fix #282 

cc @rochoa

### CODE REVIEW

File: batch/job_subscriber.js [hunk 1]
```diff
@@ -1,15 +1,60 @@
...
+        }
+
+        var queues = currentCursor[1];
+
+        if (!queues) {
+            return callback(null);
+        }
+
+        queues.forEach(function (queue) {
+            var host = queue.substr(queue.lastIndexOf(':') + 1);
```
Comment (rochoa) [inline batch/job_subscriber.js]: Isn't this the same as `this.redisPrefix.length`?

File: batch/job_subscriber.js [hunk 2]
```diff
@@ -1,15 +1,60 @@
...
+
+        self.client.subscribe(self.channel);
+        self.client.on('message', onMessage);
+    });
 };
 
+JobSubscriber.prototype._lookForQueues = function (onMessage, callback) {
+    var pattern = this.redisPrefix + '*';
+
+    scanQueues(this.metadataBackend, this.db, ['0'], pattern, this.channel, onMessage, callback);
```
Comment (rochoa) [inline batch/job_subscriber.js]: Is the `['0']` format required because how the redis client works?

File: batch/job_subscriber.js (hunk 1 above)
Comment (dgaubert) [inline batch/job_subscriber.js]: Oh! Good catch! ;)

File: batch/job_subscriber.js (hunk 2 above)
Comment (dgaubert) [inline batch/job_subscriber.js]: Yes, it is. A little weird...

**Comment by dgaubert** [comment]
I've just refactored in order to improve isolation and maintainability. Please, check it out!

**Pull request merged**